### PR TITLE
[#7] Add user data model

### PR DESCRIPTION
**Pull request by glemenneo**
Changelog: 
- Added Mongoose driver
- Add Winston logging
- Add user repository methods
- Add new DTOs and classes relating to user

### CODE REVIEW

**Review by Daviancold — COMMENTED** [review]
LGTM

File: backend/user-service/src/models/user.model.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    createdAt: {
+        type: Date,
+        required: false,
+    },
+    updatedAt: {
+        type: Date,
+        required: false,
+        default: null,
+    },
+    deletedAt: {
```
Comment (shishirbychapur) [inline backend/user-service/src/models/user.model.ts:40]: what is the purpose of storing `deletedAt` here?

**Review by shishirbychapur — APPROVED** [review]
LGTM, just curious about the `deleteAt`

File: backend/user-service/src/models/user.model.ts (hunk 1 above)
Comment (glemenneo) [reply backend/user-service/src/models/user.model.ts:40]: mostly for soft deletes and logging

Comment (shishirbychapur) [reply backend/user-service/src/models/user.model.ts:40]: okay got it, just need to handle a small edge case where the user tries to create an account with the same credentials (username, email) as before (which he deleted). (they could be rejected cuz the username/email still exists in the db)

File: backend/user-service/src/server.ts [hunk 2]
```diff
@@ -1,11 +1,38 @@
...
+    defaultMeta: { service: 'user-service' },
+    transports: [
+        new Winston.transports.Console(),
+        new Winston.transports.File({ filename: 'error.log', level: 'error' }),
+        new Winston.transports.File({ filename: 'combined.log' }),
+    ],
+})
+
 const port: string = process.env.PORT ?? '3000'
+const dbUrl: string | undefined = process.env.DB_URL
```
Comment (shishirbychapur) [inline backend/user-service/src/server.ts:20]: could you please share in the telegram group the environment variables as well, thanks!!

**Pull request merged**